core/src/game_state.rs

```diff
@@ -1139,7 +1140,12 @@ impl PlayPhase {
                 for bump_idx in 0..bump {
                     match advancement_policy {
                         // Player *must* defend on Ace and win to advance.
-                        _ if player.rank() == Number::Ace && !(is_defending && bump_idx == 0) => {
+                        _ if player.rank() == Number::Ace
+                            && !(is_defending
+                                && (!finding_friends_game
```

Comment:
I think if all of the defending team is rank A, then they should all win, no?

probably the right way to do this is to explicitly check if the landlord successfully defended A in the first loop, and then allow advances past A if the player started at A and is on the landlord's team.

This also removes the need for the check on `finding_friends_game`, since the rule would work for both tractor and finding friends

Reply:
That will block landlord with A even if the landlord wins if there is a player with non-A rank.

We can make players on landlord team with rank A to pass when landlord with A wins

Reply:
? the criteria would be

```
const landlord_successfully_defended_a = ...;

if player.rank() == Number::Ace && !(is_defending && bump_idx == 0 && landlord_successfully_defended_a) =>

```

Reply:
I could not figure out why you checked !is_defending to block advance when checking should be on is_defending. The condition I use is more intuitive I think.

Reply:
I did test playing to make sure the updated check condition works.